pennylane/ops/op_math/exp.py

```diff
@@ -179,6 +192,15 @@ def _unflatten(cls, data, metadata):
         return cls(data[0], data[1], num_steps=metadata[0])
 
     def __init__(self, base, coeff=1, num_steps=None, id=None):
+        if num_steps is not None:
+            warn(
+                "Providing 'num_steps' to 'qml.evolve' and 'qml.exp' is deprecated and will be "
+                "removed in a future version. Instead, use 'qml.TrotterProduct' for approximate "
+                "methods, providing the 'n' parameter to perform the Suzuki-Trotter product "
+                "approximation of a Hamiltonian with the specified number of Trotter steps.",
+                PennyLaneDeprecationWarning,
+            )
```

Comment:
@isaacdevlugt I updated the warning message a little bit. Let me know if it's fine or if you have any suggestions.

Reply:
```suggestion
            warn(
                "Providing 'num_steps' to 'qml.evolve' and 'qml.exp' is deprecated and will be "
                "removed in a future release. Instead, use 'qml.TrotterProduct' for approximate "
                "methods, providing the 'n' parameter to perform the Suzuki-Trotter product "
                "approximation of a Hamiltonian with the specified number of Trotter steps.",
                PennyLaneDeprecationWarning,
            )
```

This looks great! Though, I've seen a couple different conventions for how we "format" code snippets in warning messages 🤔 

1. 'qml.exp' (surrounded by single quotations)
2. qml.exp (surrounded by nothing)
3. `qml.exp` (surrounded by backticks)